### DISCUSSION TITLE
Zeta clarification

### DISCUSSION THREAD

**Discussion by rafaelmarqferreira** [oemof/tespy#532] [Q&A] [answered]
So, I´ve been wondering what exactly are the implications of the zeta value being negative, even when pressure gradients from inlet to outlet is considered to be 0. Would it be an issues with the set parameters?  How much does it affect the solution? And, most of all, what is the actual meaning of the zeta value, in practical sense?

Thanks in advance!

![image](https://github.com/user-attachments/assets/ab8cc944-6534-4eeb-8d7f-07ec6823ddda)

**Comment by fwitte** [comment, marked as answer]
Hi @rafaelmarqferreira,

the zeta value is supposed to be used in pressure loss calculation when no dimensions are available (see https://tespy.readthedocs.io/en/main/api/components.html#tespy.components.component.Component.zeta_func). Basically, the pressure delta is calculated based on flow velocity and density of the fluid. The parameter is very low (or slightly negative) with pressure change near to zero (which is the case in your calculations). The error which is prompted has no meaning in this case, and you can avoid the prompt by specifying e.g. `component.zeta.min_val = -1` (for every instance of heat exchanger you are using). The printout of the prompt is actually wrong, the minimum value should be zero (which corresponds to a pressure delta of zero). I fixed this in #529.

best

**Reply by fwitte** [reply]
... and I did a quick release with the bug fix in the logging :)
